src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModel.java

```diff
@@ -0,0 +1,269 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnlinkedFilesDialogViewModel.class);
+
+    private final ImportHandler importHandler;
+    private final StringProperty directoryPath = new SimpleStringProperty("");
+    private final ObjectProperty<FileExtensionViewModel> selectedExtension = new SimpleObjectProperty<>();
+
+    private final ObjectProperty<FileNodeViewModel> treeRootProperty = new SimpleObjectProperty<>();
+    private final SimpleListProperty<TreeItem<FileNodeViewModel>> checkedFileListProperty = new SimpleListProperty<>(FXCollections.observableArrayList());
+
+    private final BooleanProperty taskActiveProperty = new SimpleBooleanProperty(false);
+    private final DoubleProperty progressValueProperty = new SimpleDoubleProperty(0);
+    private final StringProperty progressTextProperty = new SimpleStringProperty();
+
+    private final ObservableList<ImportFilesResultItemViewModel> resultList = FXCollections.observableArrayList();
+    private final ObservableList<FileExtensionViewModel> fileFilterList;
+    private final DialogService dialogService;
+    private final PreferencesService preferences;
+    private BackgroundTask<FileNodeViewModel> findUnlinkedFilesTask;
+    private BackgroundTask<List<ImportFilesResultItemViewModel>> importFilesBackgroundTask;
+
+    private final BibDatabaseContext bibDatabase;
+    private final TaskExecutor taskExecutor;
+
+    private final FunctionBasedValidator<String> scanDirectoryValidator;
+
+    public UnlinkedFilesDialogViewModel(DialogService dialogService, ExternalFileTypes externalFileTypes, UndoManager undoManager,
+                                        FileUpdateMonitor fileUpdateMonitor, PreferencesService preferences, StateManager stateManager, TaskExecutor taskExecutor) {
+        this.preferences = preferences;
+        this.dialogService = dialogService;
+        this.taskExecutor = taskExecutor;
+        this.bibDatabase = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
+        importHandler = new ImportHandler(
+                dialogService,
+                bibDatabase,
+                externalFileTypes,
+                preferences,
+                fileUpdateMonitor,
+                undoManager,
+                stateManager);
+
+        this.fileFilterList = FXCollections.observableArrayList(
+                new FileExtensionViewModel(StandardFileType.ANY_FILE, externalFileTypes),
+                new FileExtensionViewModel(StandardFileType.BIBTEX_DB, externalFileTypes),
+                new FileExtensionViewModel(StandardFileType.PDF, externalFileTypes));
+
+        Predicate<String> isDirectory = path -> Files.isDirectory(Path.of(path));
+        scanDirectoryValidator = new FunctionBasedValidator<>(directoryPath, isDirectory,
+                ValidationMessage.error(Localization.lang("Please enter a valid file path.")));
+
+        treeRootProperty.setValue(null);
```

Comment:
You can use the EasyBind.OptionalProperty instead of handling with null values.

Reply:
No, I think it's necessary to prevent showing any empty things

Reply:
The optional property is similar to `Optional`. Just a nice wrapper so that you don't need `null` but have an `isEmpty` etc.

Reply:
I took a closer look into the `EasyBind` package, although there is an `OptionalBinding`, there seems to be no `OptionalProperty`. I took the liberty to make this an `ObjectProperty<Optional<FileNodeViewModel>>`.